core/src/main/java/com/graphhopper/routing/ch/PrepareContractionHierarchies.java

```diff
@@ -156,14 +158,31 @@ public PrepareContractionHierarchies setContractedNodes(double nodesContracted)
     }
 
     @Override
-    public void doWork() {
+    public void doSpecificWork() {
```

Comment:
I don't understand this rename :). Could you explain the background and why doSpecificWork is better than doWork (BTW: I think both names are not especially good at explaining what happens in this method)?

Reply:
The reason is highly likely that now it is not required to call the super method (so more methods but cleaner design IMO)

Reply:
Yes the idea was to be able to just override `doSpecificWork` instead of overriding `doWork` and calling `super.doWork` in the overriding method. This way AbstractAlgoPreparation is only dealing with making sure a preparation is only run once and the subclasses are only concerned with preparing the algorithm, which I think is cleaner. Regarding the names I think `doWork` isn't so bad ? Tbh I was more concerned with `PrepareContractionHierarchies` because it is not really a noun, but now I got used to it :) I would have preferred something like `ContractionHierarchiesPreparer` and then maybe naming the `doWork()` method `prepare()` instead (and maybe `doPrepare` instead of `doSpecificWork`. Then again this class not only 'prepares' the graph, but also creates the Algorithm (its a RoutingAlgorithmFactory). Anyway, I think these are all rather minor issues and I do not insist on my change if you dont like it.

Reply:
I have no preferences and the noun-argument incl. prepare sounds good.